Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -256,5 +290,17 @@ private static void HandleSessionWaitCallback(IntPtr taskCompletionSource, int e
                 tcs.TrySetException(new RealmException(OuterMessage, inner));
             }
         }
+
+        [MonoPInvokeCallback(typeof(NativeMethods.SessionPropertyChangedCallback))]
+        private static void HandleSessionPropertyChangedCallback(IntPtr propertyChangedCallbackHandle, NotifiableProperties propertyChanged)
+        {
+            ((Action<string>)GCHandle.FromIntPtr(propertyChangedCallbackHandle).Target)(propertyChanged.ToString());
+        }
+
+        // !! The names of this enum must be kept in sync with the notifiable property names of Session !!
+        private enum NotifiableProperties : byte
+        {
+            ConnectionState = 0,
+        }
```

Comment:
Rather than keep names in sync with properties, which is not something we can enforce, we should use a switch to map them like:

```csharp
var propertyName = notifiableProperty switch
{
    ConnectionState => nameof(Session.ConnectionState),
    _ => throw new NotSupportedException($"Unexpected notifiable property value: {notifiableProperty}")
};
```

Reply:
Sleek. Thank you.